state/factory/statedb.go

```diff
@@ -197,8 +197,11 @@ func (sdb *stateDB) newWorkingSet(ctx context.Context, height uint64) (*workingS
 	if err := store.Start(ctx); err != nil {
 		return nil, err
 	}
-
-	return newWorkingSet(height, store), nil
+	var parent *workingSet
+	if height > 0 {
+		parent = getWorkingSetByHeight(sdb.workingsets, height-1)
+	}
+	return newWorkingSet(height, store, parent), nil
```

Comment:
Is it only necessary to set the parent when height-1 is not yet confirmed?

Reply:
and should NOT newWorkingSet at hanging height

Reply:
yes, newWorkingSet(1) is in progress (not yet committed), and can newWorkingSet(2) at the same time